docs/notifications.md

```diff
@@ -0,0 +1,122 @@
+# Notifications
+
+Key to workflows is receiving timely notifications in the places where you work. This module provides a developer framework for creating notifications, either as hardcoded business logic or by making events, messages, recipients and destinations available to the Workflow builder UI.
+
+This component is powered by the [Workflows plugin](https://github.com/humanmade/Workflows). You can find more [detailed documentation on the plugin's wiki](https://github.com/humanmade/Workflows/wiki).
+
+## Basic hardcoded example
+
+The simplest form of a notification looks like the following:
+
+```php
+HM\Workflows\Workflow::register( 'notify_editor_on_new_post' )
+  // The action hook to trigger on.
```

Comment:
What's the "action hook" is this the transition post status name? If so, maybe we could list out the options for that here.

Reply:
That's a lot of options! The hook is `${old_status}_to_${new_status}` so I could just put that. But it can be literally any WP action called by `do_action()`